### PR TITLE
Use 'context' as the name for analyzer context parameters

### PR DESCRIPTION
**Pull request by sharwell**
The intentional aliasing ensures that context objects will not be used in the wrong scope. Examples shown in comments.

### CODE REVIEW

File: src/xunit.analyzers/TestCaseMustBeLongLivedMarshalByRefObject.cs [hunk 1]
```diff
@@ -12,21 +12,21 @@ public class TestCaseMustBeLongLivedMarshalByRefObject : XunitDiagnosticAnalyzer
...
 					return;
 
-				var semanticModel = syntaxNodeContext.SemanticModel;
+				var semanticModel = context.SemanticModel;
 				var isTestCase = false;
 				var hasMBRO = false;
 
 				foreach (var baseType in classDeclaration.BaseList.Types)
 				{
-					var type = semanticModel.GetTypeInfo(baseType.Type, compilationStartContext.CancellationToken).Type;
```
Comment (sharwell) [inline src/xunit.analyzers/TestCaseMustBeLongLivedMarshalByRefObject.cs:29]: 📝 This context object was used in the wrong scope

File: src/xunit.analyzers/TheoryMethodCannotHaveDefaultParameter.cs [hunk 2]
```diff
@@ -24,25 +24,25 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
...
 					return;
 
 				foreach (var parameter in method.Parameters)
 				{
-					symbolContext.CancellationToken.ThrowIfCancellationRequested();
+					context.CancellationToken.ThrowIfCancellationRequested();
 					if (parameter.HasExplicitDefaultValue)
 					{
 						var syntaxNode = parameter.DeclaringSyntaxReferences.First()
-							.GetSyntax(compilationStartContext.CancellationToken)
```
Comment (sharwell) [inline src/xunit.analyzers/TheoryMethodCannotHaveDefaultParameter.cs:42]: 📝 This context object was used in the wrong scope

File: src/xunit.analyzers/TheoryMethodCannotHaveParamsArray.cs [hunk 3]
```diff
@@ -23,22 +23,22 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
...
 					return;
 
 				var attributes = method.GetAttributes();
 				if (attributes.ContainsAttributeType(xunitContext.Core.TheoryAttributeType))
 				{
-					symbolContext.ReportDiagnostic(
+					context.ReportDiagnostic(
 						Diagnostic.Create(
 							Descriptors.X1022_TheoryMethodCannotHaveParameterArray,
-							parameter.DeclaringSyntaxReferences.First().GetSyntax(compilationStartContext.CancellationToken).GetLocation(),
```
Comment (sharwell) [inline src/xunit.analyzers/TheoryMethodCannotHaveParamsArray.cs:41]: 📝 This context object was used in the wrong scope

**Comment by bradwilson** [comment]
Thanks!

**Pull request merged**